core/schemas/observable.py

```diff
@@ -128,8 +102,179 @@ def delete_context(
+    if not type or type == "guess":
+        type = guess_type(value)
+        if not type:
+            raise ValueError(f"Invalid type for observable '{value}'")
+    elif type not in TYPE_MAPPING:
+        raise ValueError(f"{type} is not a valid observable type")
+    return TYPE_MAPPING[type](value=value, **kwargs)
+
+
+def save(
+    *, value: str, type: str | None = None, tags: List[str] = None, **kwargs
+) -> ObservableTypes:
+    """
+    Save an observable object. If the object is already in the database, it will be updated.
+
+    kwargs must contain a "value" field representing the of the observable.
+
+    if kwargs does not contain a "type" field, type will be automatically
+    determined based on the value. If the type is not recognized, a ValueError will be raised.
+
+    tags is an optional list of tags to add to the observable.
+    """
+    observable_obj = create(value=value, type=type, **kwargs).save()
+    if tags:
+        observable_obj.tag(tags)
+    return observable_obj
+
+
+def find(*, value, **kwargs) -> ObservableTypes:
+    return Observable.find(value=refang(value), **kwargs)
+
+
+def create_from_text(text: str) -> Tuple[List["ObservableTypes"], List[str]]:
+    """
+    Create a list of observables from a block of text.
+
+    The text is split into lines and each line is used to create an observable.
+    """
+    unknown = list()
+    observables = list()
+    for line in text.split("\n"):
+        line = line.strip()
+        if not line:
+            continue
+        try:
+            obs = create(value=line)
+            observables.append(obs)
+        except ValueError:
+            unknown.append(line)
+    return observables, unknown
+
+
+def save_from_text(
+    text: str, tags: List[str] = None
+) -> Tuple[List["ObservableTypes"], List[str]]:
+    """
+    Save a list of observables from a block of text.
+
+    The text is split into lines and each line is used to create and save an observable.
+    """
+    saved_observables = []
+    observables, unknown = create_from_text(text)
+    for obs in observables:
+        obs = obs.save()
+        if tags:
+            obs.tag(tags)
+        saved_observables.append(obs)
+    return saved_observables, unknown
+
+
+def create_from_file(file: FileLikeObject) -> Tuple[List["ObservableTypes"], List[str]]:
+    """
+    Create a list of observables from a block of text.
+
+    The text is split into lines and each line is used to create an observable.
+    """
+    opened = False
+    if isinstance(file, (str, bytes, os.PathLike)):
+        f = open(file, "r", encoding="utf-8")
+        opened = True
+    elif isinstance(file, (io.IOBase, tempfile.SpooledTemporaryFile)):
+        f = file
+    else:
+        raise ValueError("Invalid file type")
+    observables = list()
+    unknown = list()
+    for line in f.readlines():
+        if isinstance(line, bytes):
+            line = line.decode("utf-8").strip()
+        else:
+            line = line.strip()
```

Comment:
```suggestion
        if isinstance(line, bytes):
            line = line.decode("utf-8")
        line = line.strip()
```

Reply:
fixed in ef2578ac6e642a193dd0c4f67f39c69dae4766eb